cpp/src/arrow/adapters/orc/adapter.h

```diff
@@ -272,9 +272,12 @@ class ARROW_EXPORT ORCFileWriter {
       io::OutputStream* output_stream,
       const WriteOptions& write_options = WriteOptions());
 
-  /// \brief Write a table
+  /// \brief Write a table. This can be called multiple times.
   ///
-  /// \param[in] table the Arrow table from which data is extracted
+  /// Tables passed in subsequent calls must match the schema of the table that was
+  /// written first.
+  ///
+  /// \param[in] table the Arrow table from which data is extracted.
   /// \return Status
   Status Write(const Table& table);
```

Comment:
Do we similarly want to add a `Status Write(const RecordBatch& batch)` method?

Reply:
Why not, but I guess that an user can still use the Table::FromRecordBatches method to do this.
But it is up to you.

Reply:
Do you think an implementation of `Status Write(const RecordBatch& batch) `as a method that makes a Table from the RecordBatch and call the already existing method will be ok ?
Else we can do it the other way : transform tables in record batches...

Reply:
The other way would be much more logical IMHO.

Reply:
I actually think implementing `Status Write(const RecordBatch& batch)` would be more sensible here. The function `ORCFileWriter::Impl::Write()` handles the enforcement of the `batch_size` and (assuming I'm reading correctly) will handle writing batches that straddle chunked array boundaries. If you make `Status Write(const Table& table)` instead implemented by using a `TableBatchReader` and passing each individual batch to the `Write` implementation, then the writer has to break up batches based on the input chunk boundaries. IMO, ideally we should aim to decouple the chunk boundaries in-memory and the serialized chunk boundaries whenever possible.

So unless you want to do a very deep refactoring, I think we should implement `Status Write(const RecordBatch& record_batch)` by just wrapping the batch in a Table.

Reply:
Thanks for your review. I think that it would be better indeed, to implement `Status Write(const RecordBatch& record_batch)` by wrapping the batch in a table following your reasoning. I will do this, but I will first wait a bit if others want to comment.